parlai/core/torch_classifier_agent.py

```diff
@@ -213,19 +212,28 @@ def _get_labels(self, batch):
             labels_tensor = labels_tensor.cuda()
         return labels_tensor
 
-    def _update_confusion_matrix(self, batch, predictions):
+    def _update_confusion_matrix(self, predictions, batch):
         """
         Update the confusion matrix given the batch and predictions.
 
-        :param batch:
-            a Batch object (defined in torch_agent.py)
         :param predictions:
             (list of string of length batchsize) label predicted by the
             classifier
+        :param batch:
+            a Batch object (defined in torch_agent.py)
         """
-        for i, pred in enumerate(predictions):
-            label = batch.labels[i]
-            self.metrics['confusion_matrix'][(label, pred)] += 1
+        for class_name in self.class_list:
+            prec_str = 'class_{}_prec'.format(class_name)
```

Comment:
you don't need these strings anymore, right?

Reply:
Why? We still want to report that per class?

Reply:
nit: f-strings for consistency

Reply:
Thanks!